src/TestFramework/TestFramework/Attributes/TestMethod/TestCategoryAttribute.cs

```diff
@@ -18,11 +18,20 @@ public sealed class TestCategoryAttribute : TestCategoryBaseAttribute
     /// </param>
     public TestCategoryAttribute(string testCategory)
     {
-        List<string> categories =
-        [
-            testCategory
-        ];
-        TestCategories = categories;
+        // Treat empty or whitespace-only strings as null/uncategorized
+        // This ensures empty categories are not added to the list
+        if (string.IsNullOrWhiteSpace(testCategory))
```

Comment:
We should make sure to have consistent behavior in TestCategoryBaseAttribute as well.
Or instead of simply ignoring null/empty/whitespace, it should be a clear error.